smarts/core/smarts.py

```diff
@@ -994,7 +994,7 @@ def _reset_providers(self):
             except Exception as provider_error:
                 self._handle_provider(provider, provider_error)
 
-    def _handle_provider(self, provider: Provider, provider_error) -> ProviderState:
+    def _handle_provider(self, provider: Provider, provider_error):
         provider_problem = bool(provider_error or not provider.connected)
         if not provider_problem:
             return
```

Comment:
Same with for this one (`-> Optional[ProviderState]`).  But also, I think we actually should specify `return None` on line 1000.